### ISSUE TITLE
Lighting with shadows and normal maps

### ISSUE THREAD

**Issue by alberteddu** [prime31/Nez#703]
Hello! Thank you very much for making Nez. I have the most experience with unity but I really like this project and slowly getting to know it better.

I was trying to find a way to have some lights in the scene, so that normal maps from my textures are taken into account, and shadow casters can cause shadows to appear. However I have the feeling it's currently impossible to have both? 

There is a sample for deferred lighting which seems to address normal maps. There is also penumbra (for monogame) which I guess might work with Nez and addresses shadows. Finally I see a sample that uses "PolyLightPostProcessor" but I can't find the source for this class anywhere.

Do you have any pointers for me? I'm willing to do some work on this but I feel I'm lacking some knowledge at the moment.

Thanks!

**Comment by prime31** [comment]
You would need to implement some custom code for that but it is entirely possible with Nez. Nez includes a deferred Renderer, so that will get you all the normal mapped lighting. On top of that you would just need to add a shadow map.

**Comment by alberteddu** [comment]
Hey @prime31. Thank you for your answer. I know this issue is closed but I'm having some trouble implementing your suggestion. I tried but two days later I must admit there is some knowledge I am lacking that I can't gain on my own.

I studied the `DeferredLightingRenderer` you provide in the samples. I am more or less able to understand what is happening at a high level. From your suggestion, my guess is that once the Light and Normal textures are ready for the final combine, I would need to go through each "obstacle" in the scene and remove (from the light and normal textures) whatever "chunks" of light are hidden behind obstacles. 

I think I can see how this is done in `StencilLightRenderer`. In this renderer (I think) a black polygon is drawn for each light and each obstacle.

However, I am unsure how to do that in the `DeferredLightingRenderer`. Once I know which polygons of "dark areas" I need to remove from the light textures, how to modify the textures themselves?

Thank you very much for your time.
